main.go

```diff
@@ -268,18 +268,29 @@ func (ctx *httpContext) OnHttpResponseBody(bodySize int, endOfStream bool) types
 		}
 	}
 
+	// Response  body has to be buffered in order to check that it is fully legit
 	if !endOfStream {
-		return types.ActionContinue
+		return types.ActionPause
 	}
 
-	// We have already sent response headers so cannot now send an unauthorized response.
-	// The error will have been logged by Coraza though.
+	// We have already sent response headers, an unauthorized response can not be sent anymore,
+	// but we can still drop the response to prevent leaking sensitive content
+	// The error will also be logged by Coraza.
 	ctx.processedResponseBody = true
-	_, err := tx.ProcessResponseBody()
+	interruption, err := tx.ProcessResponseBody()
 	if err != nil {
 		proxywasm.LogCriticalf("failed to process response body: %v", err)
 		return types.ActionContinue
 	}
+	if interruption != nil {
+		err = proxywasm.ReplaceHttpResponseBody([]byte(``))
+		if err != nil {
+			proxywasm.LogCriticalf("failed to perform interruption on response body: %v", err)
+		}
+		// TODO(M4tteop): I'm looking for the Go version of StopIterationNoBuffer
+		// It is formally not correct to invoke SendHttpResponse during OnHttpResponseBody, but it is correctly closing the connection.
```

Comment:
what does this mean? connection of what?

Reply:
what I wanted to say is that with the current impl of envoy, it's _impossible_ to make the request fail after you return Continue in OnHttpResponseHeader https://github.com/proxy-wasm/proxy-wasm-cpp-host/issues/143

Reply:
that's the reason why we had the internal fork of cpp-host to change this behavior done by Takaya

Reply:
Thanks for confirming @mathetake. There are many proxy-wasm tasks we need to follow up on and finally have some budget for that so we'll try to make some progress there.

@M4tteoP In the meantime, it would be good to separate PRs for refactoring the example and response body logic. If you want to do the latter first, then my recommendation is to move `ProcessResponseHeaders` to `OnHttpResponseBody`). I can't find any coreruleset rules targeting response headers and this fits my intuition that there wouldn't be much to match there (maybe response cookies? But there isn't even a variable for that like there is for request cookies). There are a few response body rules though, so if we remove interruption in response headers we can still sanitize the response body when either interrupts there. What do you think?